### ISSUE TITLE
Running out of memory during building of OpenCOLLADA

### ISSUE THREAD

**Issue by danZib** [IfcOpenShell/IfcOpenShell#350]
Hi there,

I was following the step under **Compiling on nix** on the main page. As mentioned in the title, while building OpenCOLLADA on my system, I'm running out of memory. 

```
...
[ 98%] Linking CXX static library ../lib/libOpenCOLLADAStreamWriter.a
[ 98%] Built target OpenCOLLADAStreamWriter_static
make[2]: *** [COLLADASaxFrameworkLoader/CMakeFiles/OpenCOLLADASaxFrameworkLoader_static.dir/src/generated15/COLLADASaxFWLColladaParserAutoGen15Private.cpp.o] Error 4
make[2]: *** Waiting for unfinished jobs....
virtual memory exhausted: Cannot allocate memory
c++: internal compiler error: Killed (program cc1plus) 
...
```

The setup I'm using ( ubuntu server vm on mac os ) doesn't allow me to allocate more than 6 Gb of memory, which seems to be not enough for this build setup.

Is there still a way to make the build work on my system? If not, would it be possible to build on a different, more powerful machine and then moving them back to my machine?

Thank you in advance already

Daniel

**Comment by Stinkfist0** [comment]
Well that's unfortunate. Have you tried configuring swap file, maybe that would help: https://digitizor.com/create-swap-file-ubuntu-linux/

**Comment by danZib** [comment]
Thanks you @Stinkfist0 
I rerun the build process, using a swapfile of 3 GB this time. There was no error message, however after 30 min there was still no visible feedback and the build process remained at the same line. Not sure if it is likely to take that long or that the terminal just froze because of again running out of ram and swap file.

**Comment by Stinkfist0** [comment]
For the time being, you could go with the prebuilt package instead: https://github.com/KhronosGroup/OpenCOLLADA/releases 
Also, if OpenCOLLADA is not mandatory for your work, you can simply skip it (remember to pass `-DCOLLADA_SUPPORT=0` for CMake invokation).

**Comment by aothms** [comment]
Well, actually the recommended `make -j` is not very smart and can exhaust your system resources like that. You might better substitute that with `make -j5` or <number of cores> + 1.

**Comment by Stinkfist0** [comment]
Ah, good catch, please free to document that, if you have time. For OCE/OCCT `make -j` might be good default though, unless one wants to spend eternity building it. :)

**Comment by danZib** [comment]
Thank you all.

It was indeed the problem of using too many cores and hence spilling my memory. I was able to successfully build it now. Again thanks to you @Stinkfist0 and @aothms 

TL;DR: Limit number of cores during `make -j ` to avoid memory spills
